mlflow/types/schema.py

```diff
@@ -274,6 +274,15 @@ def __init__(self, inputs: List[Union[ColSpec, TensorSpec]]):
                 "Creating Schema with a combination of {0} and {1} is not supported. "
                 "Please choose one of {0} or {1}".format(ColSpec.__class__, TensorSpec.__class__)
             )
+        if (
+            all(map(lambda x: isinstance(x, TensorSpec), inputs))
+            and len(inputs) > 1
+            and any(map(lambda x: x.name is None, inputs))
+        ):
+            raise MlflowException(
+                "Creating Schema with multiple unnamed TensorSpecs is not allowed. "
+                "Please provide names for each TensorSpec or use a single unnamed TensorSpec."
```

Comment:
As this branch is only true if len(inputs)>1, informing the user to use a single TensorSpec might be confusing.
```suggestion
                "Creating Schema with multiple unnamed TensorSpecs is not supported. "
                "Please provide names for each TensorSpec."
```